bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -3433,6 +3433,7 @@ void RewriteInstance::disassembleFunctions() {
         BC->outs() << "BOLT-INFO: could not disassemble function " << Function
                    << ". Will ignore.\n";
       // Forcefully ignore the function.
+      Function.scanExternalRefs();
```

Comment:
Note that `scanExternalRefs` is now called inside `Function.setIgnored` body. It never runs twice though, due to the code structure.

I wonder if it would be clearer to perform all `scanExternalRefs` calls within `BF.setIgnored` at the relevant return points.

Maybe it's also worth adding tests for these cases using the flags `--skip-funcs=unpatchable` and ` --strict`?

Reply:
Good point. Let me think about it.

Reply:
Hey Maksim, thanks for looking into this.
Okay, let's leave it out of this patch as a lower priority.